src/PKSim.Core/Model/ParameterExtensions.cs

```diff
@@ -216,4 +204,26 @@ public static void ScaleDistributionBasedOn(this IDistributedParameter parameter
          parameter.IsFixedValue = false;
       }
    }
+
+   public static class ExpressionParameterExtensions
+   {
+      public static string ContainerNameForRelativeExpressionParameter(this ExpressionParameter expressionParameter)
+      {
+         if (expressionParameter.HasGlobalExpressionName())
+            return CoreConstants.ContainerName.GlobalExpressionContainerNameFor(expressionParameter.Name);
+         
+         var objectPath = expressionParameter.Path;
+         var containerName = parentOfParent(objectPath);
+         var key = Equals(containerName, CoreConstants.Compartment.INTRACELLULAR) ? objectPath[objectPath.Count - 4] : containerName;
+         if (expressionParameter.ContainerPath.Contains(CoreConstants.Organ.LUMEN))
+            key = CoreConstants.ContainerName.LumenSegmentNameFor(key);
+
+         return key;
+      }
+
+      private static string parentOfParent(ObjectPath objectPath)
+      {
+         return objectPath[objectPath.Count - 3];
```

Comment:
I am wondering about the 3 here. Is it possible that we have less? that 3 or 4 entries?

Reply:
Well, this code is based on similar code that uses containers to find grand parent and great grand parent. I had to modify it to use ObjectPath arrays but in the code I found, there doesn't seem to be any checks for ParentContainer == null.

I can do a count check and return the root element if there's not enough.